CONTRIBUTING.md

```diff
@@ -1,87 +1,124 @@
 
-We are very happy that you are considering implementing algorithms and data structures for others!\
-This repository is referenced and used by learners from all over the globe. Being one of our contributors, you agree and confirm that:
+## Table Of Contents
 
-- You did your work - no plagiarism allowed
-  - Any plagiarized work will not be merged.
-- Your work will be distributed under [MIT License](LICENSE.md) once your pull request is merged
-- Your submitted work fulfills our styles and standards
+* [What is an Algorithm?](#what-is-an-algorithm)
+* [Contributor agreement](#contributor-agreement)
+* [Contribution guidelines](#contribution-guidelines)
+  + [Implementation requirements](#implementation-requirements)
+  + [Nim Coding Style](#nim-coding-style)
+    - [Readability and naming conventions](#readability-and-naming-conventions)
+    - [Compilation](#compilation)
+    - [Types](#types)
+    - [Exceptions and side-effects](#exceptions-and-side-effects)
+    - [Documentation, examples and tests](#documentation-examples-and-tests)
+    - [Other](#other)
+  + [Minimal example](#minimal-example)
+  + [Submissions Requirements](#submissions-requirements)
+  + [Additional Informations](#additional-informations)
+    - [Null](#null)
 
-**New implementations** are welcome! For example, new solutions for a problem, different representations for a graph data structure, or algorithm designs with different complexity but **identical implementation** of an existing implementation is not allowed. Please check whether the solution is already implemented or not before submitting your pull request.
+## What is an Algorithm?
 
-**Improving comments** and **writing proper tests** are also highly welcome.
+An Algorithm is one or more functions that:
 
-### Contribution
+- take one or more inputs,
+- perform some internal calculations or data manipulations,
+- return one or more outputs,
+- have minimal side effects (Examples of side effects: `echo()`, `rand()`, `read()`, `write()`).
 
-We appreciate any contribution, from fixing a grammar mistake in a comment to implementing complex algorithms. Please read this section if you are contributing your work.
+## Contributor agreement
 
-Please help us keep our issue list small by adding fixes: Add the number of the issue you solved — even if only partially — to the commit message of your pull request. GitHub will use this tag to auto-close the issue when the PR is merged.
+Being one of our contributors, you agree and confirm that:
 
-#### What is an Algorithm?
+- You did your work: no plagiarism allowed. Any plagiarized work will not be merged.
```

Comment:
I stole this sentence from the Python Contributing.md file and I am not proud of it.
It does not tell which «kind» of plagiarism and is particularly vague.
I think that reimplementing from another language *is not* plagiarism (IMO, there is the translation work in that case).
Just formatting a Rosetta Code *is* plagiarism.
What if the code comes from a Nim repository? It is license-dependent and author agreement dependent too.

Reply:
Most code from RosettaCode just won't pass our code review as non-guidelines compliant :D

In my opinion we can safely delete this. If the original code was properly licensed *and* it somehow satisfies the longish list of requirements, it can be merged without any problems.

Reply:
It's removed now.